ping_delegate.go

```diff
@@ -0,0 +1,13 @@
+package memberlist
+
+import "time"
+
+// PingDelegate is used to notify an observer how long it took for a ping message to
+// complete a round trip.  It can also be used for writing arbitrary byte slices
+// into ack messages.
+type PingDelegate interface {
+	// AckPayload is invoked when an ack is being sent; the returned bytes will be appended to the ack
+	AckPayload() []byte
+	// NotifyPing is invoked when an ack for a ping is received
+	NotifyPing(other *Node, rtt time.Duration, payload []byte)
```

Comment:
I would change this to `NotifyPingComplete` since it's unclear if you are being notified that a ping is received or that a ping is complete currently.